pkg/util/deployment/deployment.go

```diff
@@ -40,6 +40,12 @@ import (
 const (
 	// The revision annotation of a deployment's replica sets which records its rollout sequence
 	RevisionAnnotation = "deployment.kubernetes.io/revision"
+	// DesiredReplicasAnnotation is the desired replicas for a deployment recorded as an annotation
+	// in its replica sets.
+	DesiredReplicasAnnotation = "deployment.kubernetes.io/desired-replicas"
```

Comment:
Ah, it is the replicas desired by the deployment.  The name is a bit confusing since Spec.Replicas is also the desired replicas.  Maybe DeploymentTotalDesiredReplicas or ParentTotalDesiredReplicas depending on if we want to reuse the annotation across controllers that manage ReplicaSets.